src/test/java/org/fundacionjala/trello/stepdefs/CardStepDefs.java

```diff
@@ -53,7 +54,53 @@ public void iCreateCardWith(final String listName, final Map<String, String> dat
      */
     @Then("I should have a card on {string} list with:")
     public void iShouldHaveACardWith(final String listName, final Map<String, String> expectedData) {
-        List<String> cardNamesList = cardPage.getAllCardNames(listName);
+        List<String> cardNamesList = cardPage.getCardNamesInList(listName);
         assertGroup.assertTrue(cardNamesList.contains(expectedData.get("name")));
     }
+
+    /**
+     * Navigates to a desired card.
+     *
+     * @param cardName expected card name to find.
+     * @param listName expected list name where to find the card.
+     */
+    @When("I navigate to the {string} card on list {string}")
+    public void navigateToCard(final String cardName, final String listName) {
+        cardPage.navigateToCard(listName, cardName);
+    }
+
+    /**
+     * Validates the card creation.
+     *
+     * @param cardTitle expected data to validate the creation.
+     */
+    @Then("I should have {string} as the card's title")
+    public void iShouldHaveCardTitle(final String cardTitle) {
+        assertEquals(cardTitle, cardPage.getCardTitle());
+    }
+
+    /**
+     * Updates a card with specific data.
+     *
+     * @param cardName expected card name to find.
+     * @param listName expected list name where to find the card.
+     * @param data     expected card data.
+     */
+    @When("I update the {string} card in the {string} list with:")
+    public void iUpdateTheCardWith(final String cardName, final String listName, final Map<String, String> data) {
+        menuBoard.closeMenuOptions();
+        form = cardPage.navigateToCard(listName, cardName);
+        form.fillForm(data);
+        form.submit();
+    }
+
+    /**
+     * Validates the card description update.
+     *
+     * @param cardDescription expected data to validate the update.
+     */
+    @Then("I should {string} as the card's description")
+    public void iShouldAsTheCardSDescription(final String cardDescription) {
+        assertEquals(cardDescription, cardPage.getCardDescription());
```

Comment:
I suggest you use SoftAssert instead.

Reply:
In this case I have just one assertion, that is why I leaved it like that